src/app/pages/ordenesdepago/components/set-movimientocontable/set-movimientocontable.component.ts

```diff
@@ -235,8 +271,27 @@ export class SetMovimientocontableComponent implements OnInit, OnDestroy {
     this.subGetNodoSeleccionadoCuenta$.unsubscribe();
   }
 
+  agregarCuentaContable1() {
+    this.subGetNodoSeleccionadoCuenta$ = this.store.select(getNodoSeleccionadoCuentaContable).subscribe((nodoCuenta: any) => {
+      if (nodoCuenta) {
+        if (Object.keys(nodoCuenta)[0] !== 'type') {
+          if (nodoCuenta && !nodoCuenta.children) {
+            this.SeleccionarCuentaContable(nodoCuenta);
+            this.cuentaContableSeleccionada1 = nodoCuenta.data;
+            nodoCuenta = null;
+            this.movimientoContable.patchValue({
+              cuentaContableEndoso: this.cuentaContableSeleccionada1
+            });
+          }
+        }
```

Comment:
Se podría simplificar...
```suggestion
      if (nodoCuenta && Object.keys(nodoCuenta)[0] !== 'type' && !nodoCuenta.children) {
        this.SeleccionarCuentaContable(nodoCuenta);
        this.cuentaContableSeleccionada1 = nodoCuenta.data;
        nodoCuenta = null;
        this.movimientoContable.patchValue({
          cuentaContableEndoso: this.cuentaContableSeleccionada1
        });
```